omegaconf/base.py

```diff
@@ -439,15 +449,22 @@ def _evaluate_complex(
         throw_on_resolution_failure: bool,
     ) -> Optional["Node"]:
         """
-        Evaluate a complex interpolation (>1, nested, with other strings...).
+        Evaluate a complex interpolation.
+
+        A complex interpolation is more elaborate than "${a}" or "${a:b,c}", e.g.:
+            "I really like ${liked}" (concatenating a string)
+            "${subject} {verb} ${object}" (concatenating multiple strings)
```

Comment:
those two are referred to everywhere else as string interpolations. This is actually user facing terminology:

see [this](https://docs.google.com/presentation/d/e/2PACX-1vT_UIV7hCnquIbLUm4NnkUpXvPEh33IKiUEvPRF850WKA8opOlZOszjKdZ3tPmf8u7hGNP6HpqS-NT5/pub?start=false&loop=false&delayms=3000&slide=id.g84632f636b_5_148).

You are adding support for a new dimension, which is nested interpolations.
this is orthogonal to simple (`${foo.bar}` or string (`${foo}_bar`)) interpolations. both can be nested or not.

Reply:
Updated docstring in 3170517

Note that `_evaluate_complex()` is now also used for string interpolations (that's why I used "complex" instead of "nested").

Reply:
Hmm, your comment about simple interpolation being potentially nested makes me think I need to to a renaming pass over the code though, because `_resolve_simple_interpolation()` doesn't support nesting. I had interpreted "simple" as "a single `${..}` block" until now. This will have to wait until next week.

Reply:
Yeah, maybe take a break and come back for a consistency pass later.